Mathlib/Analysis/Normed/Group/Lemmas.lean

```diff
@@ -1,25 +1,291 @@
 /-
-Copyright (c) 2022 Patrick Massot. All rights reserved.
+Copyright (c) 2018 Patrick Massot. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Patrick Massot
+Authors: Patrick Massot, Johannes Hölzl, Yaël Dillies
 -/
 import Mathlib.Analysis.Normed.Group.Uniform
 import Mathlib.Topology.Instances.NNReal
+import Mathlib.Topology.Instances.Rat
+import Mathlib.Topology.MetricSpace.Algebra
+import Mathlib.Topology.MetricSpace.IsometricSMul
+import Mathlib.Topology.Sequences
+
+#align_import analysis.normed.group.basic from "leanprover-community/mathlib"@"41bef4ae1254365bc190aee63b947674d2977f01"
 
 /-!
-# Further lemmas about normed groups
+# Normed (semi)groups
+
+In this file we define 10 classes:
 
-This file contains further lemmas about normed groups, requiring heavier imports than
-`Mathlib/Analysis/Normed/Group/Basic.lean`.
+* `Norm`, `NNNorm`: auxiliary classes endowing a type `α` with a function `norm : α → ℝ`
+  (notation: `‖x‖`) and `nnnorm : α → ℝ≥0` (notation: `‖x‖₊`), respectively;
+* `Seminormed...Group`: A seminormed (additive) (commutative) group is an (additive) (commutative)
+  group with a norm and a compatible pseudometric space structure:
+  `∀ x y, dist x y = ‖x / y‖` or `∀ x y, dist x y = ‖x - y‖`, depending on the group operation.
```

Comment:
I think the former module doc was more accurate?

Reply:
Oops, I meant to fix that. Thanks for reminding me